### PR TITLE
Support different flavours of packages

### PR DESCRIPTION
**Pull request by jacobstanley**
This allows the profiling build or package documentation to be installed
when required, the package is "upgraded" in place in the global cache.

### CODE REVIEW

File: src/Mafia/Init.hs [hunk 1]
```diff
@@ -150,18 +154,16 @@ data Profiling =
...
 
-data InstallState =
-  InstallState {
-      isSourceDependencies :: Set SourcePackage
-    , _isCabalFile         :: Hash
-    } deriving (Eq, Ord, Show)
-
 data MafiaState =
   MafiaState {
-      msInstallState :: InstallState
```
Comment (charleso) [inline src/Mafia/Init.hs:161]: Any reason for this change apart from cleaning up? Looks like the json file has the same information?

File: src/Mafia/Install.hs [hunk 2]
```diff
@@ -163,50 +181,96 @@ link w db env p@(Package (PackageRef pid _ _) _ _) = do
...
+
+flavourArgs :: Flavour -> [Argument]
+flavourArgs = \case
+  Vanilla ->
+    []
+  Profiling ->
+    [ "--reinstall"
+    , "--ghc-options=-fprof-auto-exported"
+    , "--enable-library-profiling" ]
+  Documentation ->
```
Comment (charleso) [inline src/Mafia/Install.hs:232]: I guess at some point we'll want to call this again

**Comment by charleso** [comment]
:+1:

File: src/Mafia/Init.hs (hunk 1 above)
Comment (jacobstanley) [inline src/Mafia/Init.hs:161]: yeah just cleaning up

Comment (jacobstanley) [inline src/Mafia/Init.hs:161]: whether profiling is installed is morally part of the install state, which is just the mafia state now

bumping the version on the json state is basically free, recreating it only takes 5-10secs

File: src/Mafia/Install.hs (hunk 2 above)
Comment (jacobstanley) [inline src/Mafia/Install.hs:232]: yeah i think we'd call that when someone asks for hoogle or for haddock docs at some point

**Comment by jacobstanley** [comment]
@charleso can you take a look at the new "Install hackage packages" commit

File: src/Mafia/Install.hs [hunk 3]
```diff
@@ -163,50 +181,108 @@ link w db env p@(Package (PackageRef pid _ _) _ _) = do
...
+  ignoreIO (removeDirectoryRecursive sbdir)
+  createDirectoryIfMissing True sbdir
+
+  Hush <- sbcabal "sandbox" ["init", "--sandbox", sbdir]
+
+  case msrc of
+    -- hackage package
+    Nothing -> do
+      createDirectoryIfMissing True sbsrc
+      Hush <- sbcabal "unpack" ["--destdir=" <> sbsrc, renderPackageId pid]
```
Comment (charleso) [inline src/Mafia/Install.hs:265]: A comment for why you do this might be good - it won't be obvious over time :)

**Comment by charleso** [comment]
:+1:

**Pull request merged**